### PR TITLE
guide for jbang integration

### PR DESCRIPTION
**Pull request by maxandersen**
follow up for #11542

this is a basic example based on the rest getting started guide but just done via jbang instead + showing the few gotchas/differences with current jbang (i.e. single source, Q:config vs application.properties, etc.)

### CODE REVIEW

File: docs/src/main/asciidoc/scripting.adoc [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+
+First, we need a java file. jbang lets you create an initial version using:
+
+[source,shell,subs=attributes+]
+----
+jbang scripting/quarkusapp.java
+cd scripting
+----
+
+This command generates that you can directly run on Linux and OS X, i.e. `./quarkusapp.java` - on Windows you need to use `jbang quarkusapp.java`
```
Comment (gastaldi) [inline docs/src/main/asciidoc/scripting.adoc:94]:
```suggestion
This command generates a .java file that you can directly run on Linux and OS X, i.e. `./quarkusapp.java` - on Windows you need to use `jbang quarkusapp.java`
```

File: docs/src/main/asciidoc/scripting.adoc [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+This line is what on Linux and OS X allows you to run it as a script. On Windows this line is ignored.
+
+The next line
+
+[source,java]
+----
+// //DEPS <dependency1> <dependency2>
+----
+
+Is illustrating how you add dependencies to this script. This ia feature of `jbang`.
```
Comment (gastaldi) [inline docs/src/main/asciidoc/scripting.adoc:116]:
```suggestion
Is illustrating how you add dependencies to this script. This is a feature of `jbang`.
```

File: docs/src/main/asciidoc/scripting.adoc [hunk 3]
```diff
@@ -0,0 +1,440 @@
...
+
+[TIP]
+.How do I edit this file and get content assist ?
+====
+As there is nothing but a `.java` file most IDE's does not handle content assist well.
+To work around that you can run `jbang edit quarkusapp.java` which will print out a directory that will have a temporary project setup you can use in your IDE.
+
+On Linux/OSX you can run `<idecommand> `jbang edit quarkusapp.java``.
+
+If you add dependenciees while editing you can get jbang to automatically refresh
```
Comment (gastaldi) [inline docs/src/main/asciidoc/scripting.adoc:153]:
```suggestion
If you add dependencies while editing you can get jbang to automatically refresh
```

File: docs/src/main/asciidoc/scripting.adoc [hunk 4]
```diff
@@ -0,0 +1,440 @@
...
+== Using injection
+
+Dependency injection in Quarkus is based on ArC which is a CDI-based dependency injection solution tailored for Quarkus' architecture.
+You can learn more about it in the link:cdi-reference[Contexts and Dependency Injection guide].
+
+ArC comes as a dependency of `quarkus-resteasy` so you already have it handy.
+
+Let's modify the application and add a companion bean.
+
+Normally you would add a seperate class, but as we are aiming to have it all in one file you will add a
```
Comment (gastaldi) [inline docs/src/main/asciidoc/scripting.adoc:248]:
```suggestion
Normally you would add a separate class, but as we are aiming to have it all in one file you will add a 
```

File: docs/src/main/asciidoc/scripting.adoc [hunk 5]
```diff
@@ -0,0 +1,440 @@
...
+}
+----
+
+[TIP]
+.Use of nested static public classes
+====
+We are using a nested static public class instead of a top level class for two reasons:
+
+  1. jbang currently does not currently support multiple source files
+  2. All java frameworks relying on introspection have challenges using top level classes as they are not as visible as public classes; and in java there can only be one top level public class in a file.
```
Comment (gastaldi) [inline docs/src/main/asciidoc/scripting.adoc:271]:
```suggestion
  2. All Java frameworks relying on introspection have challenges using top level classes as they are not as visible as public classes; and in java there can only be one top level public class in a file.
```

File: docs/src/main/asciidoc/scripting.adoc [hunk 6]
```diff
@@ -0,0 +1,440 @@
...
+
+[source,shell,subs=attributes+]
+----
+$ curl -w "\n" http://localhost:8080/hello/greeting/quarkus
+hello null
+----
+
+Now that is unexpected, why is it returning `hello null` and not `hello quarkus` ?
+
+The reason is that JAX-RS `@PathParam` relies on the `-parameters` compiler flag to be set to be able to map `{name}` to the `name` parameter.
```
Comment (gastaldi) [inline docs/src/main/asciidoc/scripting.adoc:334]: Only in Quarkus or is that per spec?

File: docs/src/main/asciidoc/scripting.adoc [hunk 7]
```diff
@@ -0,0 +1,440 @@
...
+//JAVA_OPTIONS -Djava.util.logging.manager=org.jboss.logmanager.LogManager
+----
+
+With that in place running `jbang quarkusapp.java` will log and render as expected.
+
+== Configuring Application
+
+You can use `//Q:CONFIG <property>=<value>` to setup static configuration for your application.
+
+i.e. if you wanted to add the `smallrye-openapi` and `swagger-ui` and have the swagger-ui always show up you would add the foolowing:
```
Comment (gastaldi) [inline docs/src/main/asciidoc/scripting.adoc:380]:
```suggestion
i.e. if you wanted to add the `smallrye-openapi` and `swagger-ui` and have the swagger-ui always show up you would add the following:
```

File: docs/src/main/asciidoc/scripting.adoc [hunk 8]
```diff
@@ -0,0 +1,440 @@
...
+ -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \
+--\___\_\____/_/ |_/_/|_/_/|_|\____/___/
+2020-08-30 06:23:36,846 INFO  [io.quarkus] (main) Quarkus {quarkus-version} native started in 0.015s. Listening on: http://0.0.0.0:8080
+2020-08-30 06:23:36,846 INFO  [io.quarkus] (main) Profile prod activated.
+2020-08-30 06:23:36,846 INFO  [io.quarkus] (main) Installed features: [cdi, resteasy]
+----
+
+=== Conclusion
+
+If you want to get started with Quarkus or write something quickly Quarkus Scripting with jbang lets you do that. No maven, no gradle - just a java file. In this guide we outlined the very basics on using Quarkus with jbang; if you want to learn more about what jbang can do go see https://jbang.dev.
```
Comment (gastaldi) [inline docs/src/main/asciidoc/scripting.adoc:439]:
```suggestion
If you want to get started with Quarkus or write something quickly, Quarkus Scripting with jbang lets you do that. No Maven, no Gradle - just a java file. In this guide we outlined the very basics on using Quarkus with jbang; if you want to learn more about what jbang can do go see https://jbang.dev.
```

**Review by gastaldi — CHANGES_REQUESTED** [review]
Found some typos, looks good

File: docs/src/main/asciidoc/scripting.adoc (hunk 6 above)
Comment (maxandersen) [reply docs/src/main/asciidoc/scripting.adoc:334]: Per spec. 

Openjdk didn't enable it by default for whatever reasons :)

**Comment by maxandersen** [comment]
merged changes and squashed!

**Review by gastaldi — APPROVED** [review]
LGTM

**Pull request merged**